base/channels.jl

```diff
@@ -129,12 +135,13 @@ julia> String(collect(chnl))
 "hello world"
 ```
 """
-function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false) where T
+function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false, threadpool=:default) where T
```

Comment:
Also also: do we need to do error-checking for other flag values besides `:default` and `:interactive`? or does that happen inside _spawn_set_thrpool?

Reply:
Yes that function will check it.